weave-js/src/components/Panel2/PanelPanel.tsx

```diff
@@ -220,119 +243,92 @@ export const useUpdateServerPanel = (
   return updateConfigForPanelNode;
 };
 
-const usePanelPanelCommon = (props: PanelPanelProps) => {
-  const weave = useWeaveContext();
-  const {updateInput} = props;
-  const updateConfig2 = useUpdateConfig2(props);
-  const panelQuery = CGReact.useNodeValue(props.input);
-  const selectedPanel = useSelectedPath();
-  const setSelectedPanel = useSetInspectingPanel();
-  // const panelConfig = props.config;
+interface PanelRootContextState {
+  state: PanelRootsState;
+  dispatch: React.Dispatch<ActionWithId>;
+}
+
+const PanelRootContext = React.createContext<PanelRootContextState | null>(
+  null
+);
+PanelRootContext.displayName = 'PanelRootContext';
+
+export const PanelRootContextProvider: React.FC<{}> = props => {
```

Comment:
We might want to add a safe gaurd/check to make sure this is never nested inside of another instance of such provider.

Reply:
actually, i think it might be more subtle... we might want to say: if `PanelRootContextProvider` is a child of another `PanelRootContextProvider`, then this is a no-op. This way things like reports (that render configs on behalf of panels) can have a higher-level provider...

Reply:
This is intended to be used on a page. I think we should throw if there's a nested one.